launch_ros/launch_ros/actions/node.py

```diff
@@ -0,0 +1,184 @@
+# Copyright 2018 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module for the Node action."""
+
+import logging
+from typing import Dict
+from typing import Iterable
+from typing import List
+from typing import Optional
+from typing import Text
+
+from launch.action import Action
+from launch.actions import ExecuteProcess
+from launch.launch_context import LaunchContext
+from launch.some_substitutions_type import SomeSubstitutionsType
+from launch.substitutions import LocalSubstitution
+from launch.utilities import normalize_to_list_of_substitutions
+from launch.utilities import perform_substitutions
+
+from launch_ros.substitutions import ExecutableInPackage
+
+from rclpy.validate_namespace import validate_namespace
+from rclpy.validate_node_name import validate_node_name
+
+_logger = logging.getLogger(name='launch_ros')
+
+
+class Node(ExecuteProcess):
+    """Action that begins executing a process and sets up event handlers for the process."""
+
+    def __init__(
+        self, *,
+        package: SomeSubstitutionsType,
+        node_executable: SomeSubstitutionsType,
+        node_name: Optional[SomeSubstitutionsType] = None,
+        node_namespace: Optional[SomeSubstitutionsType] = None,
+        remappings: Optional[Dict[SomeSubstitutionsType, SomeSubstitutionsType]] = None,
```

Comment:
Aren't there cases where the specific order of the remapping rules has to be preserved, because applying them in different orders causes different results? If so, another type would be more appropriate e.g. OrderedDict or list of tuples.

Reply:
That's true, let me look into that.

Reply:
So, I can't use OrderedDict as a type annotation with subscripted types for key and value, unfortunately. I don't like using a list of tuples because of how it looks/feels in the examples, but I don't see another way, and creating an ordered dict also uses a list of tuples. So I guess I'll do that, but the demos/examples need to be updated.

Reply:
313bd70